include/deal.II/grid/grid_generator.h

```diff
@@ -1746,6 +1763,102 @@ namespace GridGenerator
+   * of the surface mesh facets.
+   * - CGAL::parameters::facet_size: a scalar field (resp. a constant)
+   * describing a space varying (resp. a uniform) upper-bound or for the radii
+   * of the surface Delaunay balls.
+   * - CGAL::parameters::facet_distance: a scalar field (resp. a constant)
+   * describing a space varying (resp. a uniform) upper bound for the distance
+   * between the facet circumcenter and the center of its surface Delaunay ball.
+   * - CGAL::parameters::facet_topology: the set of topological constraints
+   * which have to be verified by each surface facet. The default value is
+   * CGAL::FACET_VERTICES_ON_SURFACE. See CGAL::Mesh_facet_topology manual page
+   * to get all possible values.
+   * - CGAL::parameters::cell_radius_edge_ratio: an upper bound for the
+   * radius-edge ratio of the mesh tetrahedra.
+   * - CGAL::parameters::cell_size: a scalar field (resp. a constant) describing
+   * a space varying (resp. a uniform) upper-bound for the circumradii of the
+   * mesh tetrahedra.
+   *
+   * In dimension two only three parameters are accepted:
+   * - angular_bound is a lower bound in degrees for the angles of mesh facets.
+   * - radius_bound is an upper bound on the radii of surface Delaunay balls. A
+   * surface Delaunay ball is a ball circumscribing a mesh facet and centered on
+   * the surface.
+   * - distance_bound is an upper bound for the distance between the
+   * circumcenter of a mesh facet and the center of a surface Delaunay ball of
+   * this facet.
+   *
+   * An example usage of this function is given by
+   *
+   * @code
+   * Triangulation<dim, 3>  tria;
+   * FunctionParser<3> my_function("(1-sqrt(x^2+y^2))^2+z^2-.25");
+   * GridGenerator::implicit_function( tria, my_function,
+   *      Point<3>(.5, 0, 0), 1.0, cell_size = 0.2);
+   * @endcode
+   *
+   * The above snippet of code generates the following grid for `dim` equal to
+   * two and three respectively
+   *
+   * @image html grid_generator_implicit_function_2d.png
+   *
+   * @image html grid_generator_implicit_function_3d.png
+   *
+   * @ingroup simplex
+   *
+   * @param[out] tria The output triangulation
+   * @param[in] implicit_function The implicit function
+   * @param[in] interior_point A point in the interior of the domain, for which
+   * @p implicit_function is negative
+   * @param[in] outer_ball_radius The radius of the ball that will contain the
+   * generated Triangulation object
+   * @param[in] cgal_args Additional parameters to pass to the CGAL::make_mesh_3
+   * function and to the CGAL::make_surface_mesh functions
+   */
+  template <int dim, typename... Args>
+  void
+  implicit_function(Triangulation<dim, 3> &tria,
+                    const Function<3> &    implicit_function,
+                    const Point<3> &       interior_point    = Point<3>(),
+                    const double &         outer_ball_radius = 1.0,
+                    Args... cgal_args);
```

Comment:
I am not a huge fan of these variadic arguments. For one, they are not named, and they don't have a type. As a consequence, one has no idea what could possibly be passed here, and there is no error checking at the call site whether they make sense. Their order is also unspecified.

I would very much prefer there to be a `struct` of some kind that encapsulates and names what arguments one can pass here. (If these arguments are just passed through to some other function within deal.II, then the same is true for these as well.)